zcash_primitives/src/note_encryption.rs

```diff
@@ -531,7 +528,7 @@ pub fn try_sapling_output_recovery(
         return None;
     }
 
-    let to = PaymentAddress { pk_d, diversifier };
+    let to = PaymentAddress::from_parts(diversifier, pk_d)?;
```

Comment:
This can propagate the error by returning `None`, but that is correct and fixes a bug :-)